src/_mssql.pyx

```diff
@@ -743,17 +743,29 @@ cdef class MSSQLConnection:
         with nogil:
             dbclose(self.dbproc)
 
-        self.mark_disconnected()
+        self._real_close()
 
-    def mark_disconnected(self):
-        log("_mssql.MSSQLConnection.mark_disconnected()")
-        self.dbproc = NULL
+    def _real_close(self):
+        """
+        Free resources used by this object.
+        """
+        log("_mssql.MSSQLConnection.real_close()")
+        if self in connection_object_list:
```

Comment:
Yes, it's a `def` method so any Python construct is fair game.

(I've tested it and Cython builds it withour errors). See also http://docs.cython.org/src/userguide/language_basics.html#python-functions-vs-c-functions